### PR TITLE
Iteration1 dev fs

### PR DESCRIPTION
**Pull request by francisco-springer**
creating pull request to make it easier to comment on changes.

### CODE REVIEW

File: StudyBoosterEJB/pom.xml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		<!-- <artifactId>javaee-web-api</artifactId> -->
+		<!-- </dependency> -->
+		<dependency>
+			<groupId>javax</groupId>
+			<artifactId>javaee-api</artifactId>
+			<scope>provided</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>StudyBooster.com</groupId>
```
Comment (navteniev) [inline StudyBoosterEJB/pom.xml]: This dependency is missing from the pull request.  Since it's missing, maven can't build this module and also can't build the parent module.  The issue is that the StudyBoosterJPA is not listed as a module in the top level pom.xml.  Adding `<module>StudyBoosterJPA</module>` to the top level pom enables the maven project to build.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/Event.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+import java.util.List;
+
+
+import javax.inject.Named;
+
+@Named("event")
+public class Event 
+{
+	int eventID;
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/Event.java]: All these fields are currently package-private which would allow other classes to access these fields and bypass the getter and setter methods available for this class.  What is the reason for making these fields package-private?

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+public class EventEngine
+{
+	List<Event> eventList;
+
+	List<EventTopic>eventTopicList;
+	
+	int eventTopicId;
+	
+	@Inject @Named("myRand")
+	private  MyRand myRand;
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: Injecting MyRand into a private instance field will make this class challenging to unit test.  I would suggest injecting this via a constructor.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+			
+			eventList.add(event);
+		}
+
+		return eventList;
+	}
+	
+	private void generateEventTopicList()
+	{
+		eventTopicList = new ArrayList<EventTopic>();
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: This instance variable is being used to pass the list between this function and the client.  Since the instance variable is not being used by another method I would suggest changing it to be a local variable and modifying the method signature to return a List of generated event topics.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+		int topicLen = myRand.randNum(1, 5);
+		
+		for (int x = 0; x < topicLen; x++)
+		{
+			int len = myRand.randNum(5, 10);
+			String eventTopicTitle = myRand.randomString(len);
+			len = myRand.randNum(5, 15);
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: Reusing a variable in this way might make it difficult to make code changes to this function in the future.  since the value of len is only used as an argument to the randomString function would you consider inlining the method call to generate the value instead of assigning it to the local variable?

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+			String eventTopicSubtitle = myRand.randomString(len);
+
+			EventTopic eventTopic = new EventTopic(eventTopicId, eventTopicTitle, eventTopicSubtitle);
+
+			eventTopicList.add(eventTopic);
+
+			eventTopicId++;
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: I think I understand what this is trying to do, which is generate incremental event topic IDs.  It's hard to follow this since the value is reset in the `generateEventList` method but incremented in the body of this method.  I think the same functionality can be implemented without using an instance variable, what do you think are the benefits to using an instance variable here?

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import com.StudyBooster.Misc.MyRand;
+
+@Named("eventEngine")
+public class EventEngine
+{
+	List<Event> eventList;
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: It looks like all the fields are package-private in this module.  Would you be willing to explain the reason for that?

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Ejbs/EventEjb.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import com.StudyBooster.Biz.Event;
+import com.StudyBooster.Biz.EventEngine;
+
+@Stateless
+public class EventEjb implements Serializable
+{
+	private static final long serialVersionUID = 1L;
+
+	@Inject
+	private @Named("eventEngine") EventEngine eventEngine;
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Ejbs/EventEjb.java]: I'd suggest injecting this in a constructor to enable unit testing.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
+package com.StudyBooster.Misc;
+
+import java.sql.Time;
+import java.util.GregorianCalendar;
+import java.util.Random;
+
+import javax.inject.Named;
+
+@Named("myRand")
+public class MyRand
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: General comment about this class.  It seems to be here to help generate fake data.  Since that is the case, I'd suggest using an existing tool instead of writing something from scratch, maybe something like this: http://dius.github.io/java-faker/

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+{
+
+	String SALTCHARS;
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: This is a constant it can be made private, static, and initialized during construction if you choose to keep this class at all.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+	public MyRand()
+	{
+		SALTCHARS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
+	}
+
+	public String randomString(int lenght)
+	{
+		StringBuilder eventProperty = new StringBuilder();
+		Random rnd = new Random();
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: Since this is a class with instance variables I think creating an instance of `Random` and keeping a reference to it in the field will result in code that is a bit more efficient.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+	{
+		return (new Random()).nextInt((max - min) + 1) + min;
+		// return 5;
+	}
+
+	public String randomTime()
+	{
+		final Random random = new Random();
+		final int millisInDay = 24 * 60 * 60 * 1000;
+		Time time = new Time((long) random.nextInt(millisInDay));
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: Time expects an argument which is 

> milliseconds since January 1, 1970, 00:00:00 GMT; 

This method looks like it is assuming that the expectation is the time in milliseconds since the start of the current day.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+		return time.toString();
+
+	}
+	
+	@SuppressWarnings("static-access")
```
Comment (navteniev) [inline StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: It looks like this implementation is using some constants defined in `GergorianCalendar.`  What is the benefit to suppressing the warnings here instead of accessing the class variables or using static import?

File: StudyBoosterEJB/src/test/java/com/StudyBooster/Test/EventTest.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class EventTest
```
Comment (navteniev) [inline StudyBoosterEJB/src/test/java/com/StudyBooster/Test/EventTest.java]: This test is not testing any properties of the event.  It looks like it's here simply to show that unit testing is set up, is that the case?

File: StudyBoosterWeb/src/main/java/com/StudyBooster/Requests/EventComponent.java [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+	
+	@Inject @Named("eventEngine")
+	private  EventEngine eventEngine;
+	
+	private  List<Event> eventList;
+
+	public EventComponent(){}
+	
+	@PostConstruct
+	public void CreateEventComponent()
```
Comment (navteniev) [inline StudyBoosterWeb/src/main/java/com/StudyBooster/Requests/EventComponent.java]: this method name violates java naming conventions, is this a special naming convention for EBJ?

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/Event.java (hunk 2 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/Event.java]: This was a temporary class i created to load data into the landing page cards and get started with the website. I got rid of this class all together since this class became an entity now.
If i recall correctly Sir, glassfish would complain and not deploy the site if i did not make this field private and the class was being injected some where in the code.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java (hunk 3 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: Sir, I will take this into consideration. However, this code is not needed any longer since it was being used by the Event class to populate the event data on the cards which was randomly generated data. This code actually moved up into a higher hierarchy, the db, which I am currently using to initialize and populate my database. I hooked up the db into the website and all data is fed from the database and not from temporary java code any longer. 

I was able to compile and deploy this code into glassfish and did not have any problems with it Sir.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java (hunk 4 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: Sir, EventEngine class has been rewritten. Before, it was generating random strings to populate cards on the landing page. I did this to get the website started with data. This is working from the db now and returns an Event entity that contains EventTopics instances. However, the current method, that returns Events entities, is following this suggestion.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java (hunk 5 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: Sir, this is no longer part of the code any more, sorry, it was temp code to get started. However, I will try to remember for future reference.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java (hunk 6 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: That is correct Sir, it is generating incremental values for eventtopics.  
It was easy to write. 
How would you have done it differently?

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java (hunk 7 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Biz/EventEngine.java]: Sir, if i remember correctly, glassfish would not deploy the project unless these fields are private and accessed through their public getters and setters only for classes that are injected into code.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java (hunk 9 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: thank you Sir, I will keep this in mind for future use.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java (hunk 10 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: got it Sir, but, this class is out the window now. thanks.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java (hunk 11 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: I see Sir, that way the injected class does not have to redefine the rnd object in each method, cool.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java (hunk 12 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: Yes Sir, it takes the number of milli seconds that are in a day as the seed so that it would return a milli second between 0 and 24 * 60 * 60 * 1000, which is the parameter of Time function.

File: StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java (hunk 13 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/main/java/com/StudyBooster/Misc/MyRand.java]: Sir, I thought it was the easiest solution.

File: StudyBoosterEJB/src/test/java/com/StudyBooster/Test/EventTest.java (hunk 14 above)
Comment (francisco-springer) [reply StudyBoosterEJB/src/test/java/com/StudyBooster/Test/EventTest.java]: Yes Sir, is a template for me to use.

File: StudyBoosterWeb/src/main/java/com/StudyBooster/Requests/EventComponent.java (hunk 15 above)
Comment (francisco-springer) [reply StudyBoosterWeb/src/main/java/com/StudyBooster/Requests/EventComponent.java]: No Sir, it is not. I will follow java naming conventions.

If it is Ok with you, i will follow the below naming convention

class name => should start with uppercase letter 

interface name => should start with uppercase letter 

method name => should start with lowercase letter 

variable name => should start with lowercase letter 

package name => should be in lowercase letter 

constants name => should be in uppercase letter

However, I will make all upper caseentity classes. This is a mariadb thing so that jpa/provider can find the tables. 

Thank you Sir.
